dashboard/app/models/user.rb

```diff
@@ -830,6 +831,26 @@ def update_with_password(params, *options)
     end
   end
 
+  def update_primary_authentication_option(email)
+    # If an auth option exists with same email, set it to the user's primary authentication option
+    existing_auth_option = authentication_options.find {|ao| ao.email == email}
+    if existing_auth_option
+      self.primary_authentication_option = existing_auth_option
+      return save
+    end
+
+    # If an email option with a different email address already exists, destroy it
+    existing_email_option = authentication_options.find {|ao| ao.credential_type == 'email'}
+    existing_email_option&.destroy
+
+    self.primary_authentication_option = AuthenticationOption.new(
+      email: email,
+      credential_type: 'email',
+      user: self
+    )
```

Comment:
Is the `AuthenticationOption` persisted by the `User` `save` call?

Reply:
yep, when the user saves it saves relational data too